### PR TITLE
Tessen setup + utilities

### PR DESCRIPTION
**Pull request by jerelmiller**
Relates to https://github.com/newrelic/docs-website/issues/789

[README updates](https://github.com/newrelic/gatsby-theme-newrelic/tree/jerel/tessen/packages/gatsby-theme-newrelic)

# Description

Adds the ability to load tessen on to a site using tessen configuration. Also provides a `useTesen` hook that can be used to instrument with Tessen, and the ability to auto-instrument page views with tessen.

### CODE REVIEW

**Review by zstix — DISMISSED** [review]
Code-level stuff looks good. I like the choices you've made around organizing this stuff!

**Comment by caylahamann** [comment]
qq: have you been able to test that this reports to tessen? (I remember something being said about tessen actions not reporting or being weird)

**Comment by jerelmiller** [comment]
@caylahamann I have not. I didn't know how best to test this without skewing actual data. I can use the write key from the doc site, but I didn't want to create false data.

**Comment by jerelmiller** [comment]
I have verified that I can at least get access to a `Tessen` global which I can then call methods on.

File: packages/gatsby-theme-newrelic/README.md [hunk 1]
```diff
@@ -2463,6 +2542,52 @@ const SearchInput = () => {
...
+  page: (name: string, category: string, properties?: object) => void;
+  track: (name: string, category: string, properties?: object) => void;
+};
+```
+
+**Examples**
+
+```js
+const MyComponent = () => {
+  const tessen = useUserId();
```
Comment (caylahamann) [inline packages/gatsby-theme-newrelic/README.md:2581]:
```suggestion
  const tessen = useTessen();
```

**Review by caylahamann — APPROVED** [review]
Looks all good to me! the only thing is one minor typo

File: packages/gatsby-theme-newrelic/README.md (hunk 1 above)
Comment (jerelmiller) [reply packages/gatsby-theme-newrelic/README.md:2581]: Oops good catch! Now you see my copy/paste secret 😜

**Review by caylahamann — APPROVED** [review]

**Pull request merged**